url-shortener/backend/src/Backend.hs

```diff
@@ -0,0 +1,58 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeFamilies #-}
+module Backend where
+
+import qualified Common.Route as R
+import Control.Monad.IO.Class (liftIO)
+import qualified Data.Aeson as A
+import Data.ByteString.Lazy (toStrict)
+import Data.Text (Text)
+import Data.Word (Word64)
+import Database.Id.Class (unId)
+import Data.Pool (withResource)
+import Database.PostgreSQL.Simple (execute, Query, query, query_)
+import Gargoyle.PostgreSQL.Connect (withDb)
+import Obelisk.Backend (Backend(..), _backend_run, _backend_routeEncoder)
+import Obelisk.Route (pattern (:/))
+import qualified Snap.Core as S
+
+maxUrlSize :: Word64
+maxUrlSize = 2000
+
+migration :: Query
+migration = "CREATE TABLE IF NOT EXISTS urls\
+  \ (id SERIAL PRIMARY KEY, url VARCHAR(?) NOT NULL);"
```

Comment:
max char length has no benefit in postgres and also adds some complexity to this project that probably isn't worth it. I'd say drop this. We can still check max URL size on the request end.